### PR TITLE
155-overlay_segmentation-function-is-missing

### PR DESCRIPTION
**Pull request by Pfleiderer-Adrian**
- add missing "overlay_segmentation"-function
- adjust function calls with necessary num_classes-value
- add seaborn package for color palette generation
- add seaborn package in the requirements.txt file

### CODE REVIEW

**Comment by muellerdo** [comment]
seaborn needs to be added to setup.py as well

**Comment by Pfleiderer-Adrian** [comment]
> seaborn needs to be added to setup.py as well

done.

**Comment by Deathlymad** [comment]
So afaik the overlay_segmentation was renamed to overlay_segmentation_greyscale. The reason for that was the the existing overlay segmentation function did not handle RGB valued volume inputs that may exist. 

It would be great if you pointed the motivation out for me again. (What overlay_segmentation has over overlay_segmentation_greyscale)
Furthermore it would be great if you removed the need for seaborn. Matplotlib can handle colormaps in a similar fashion which would remove the need for the library.

**Comment by Pfleiderer-Adrian** [comment]
> So afaik the overlay_segmentation was renamed to overlay_segmentation_greyscale. The reason for that was the the existing overlay segmentation function did not handle RGB valued volume inputs that may exist.
> 
oh ok. now i understand why that was renamed. I will revise the pull request again for RGB volumes.

> It would be great if you pointed the motivation out for me again. (What overlay_segmentation has over overlay_segmentation_greyscale) Furthermore it would be great if you removed the need for seaborn. Matplotlib can handle colormaps in a similar fashion which would remove the need for the library.

First, the function calls have not been adjusted. All function-calls still call the old "overlay_segmentation" function. This results in a function not found error. Furthermore, the overlay_segmentation_greyscale function generates extremely dark colors (black, gray, ..) for ascending labels (0, 1, 2, ..) which cannot be recognized on greyscale images (only tested on master-branch maybe this has changed in dev). With the color_palette-function, Seaborn delivers a uniform and ideally distributed color spectrum across all classes to ensure that the colors are as different as possible for each class.

**Comment by Deathlymad** [comment]
https://github.com/frankkramer-lab/MIScnn/pull/156/commits/c3736a109a511f0dfc68166edf5a12ad74744238#diff-1da7b73038d32317489f303c1059e11843a1b7235177fe2e0c2e6566057b72f7R167-R168
https://github.com/frankkramer-lab/MIScnn/pull/156/commits/c3736a109a511f0dfc68166edf5a12ad74744238#diff-1da7b73038d32317489f303c1059e11843a1b7235177fe2e0c2e6566057b72f7R186
https://github.com/frankkramer-lab/MIScnn/pull/156/commits/c3736a109a511f0dfc68166edf5a12ad74744238#diff-1da7b73038d32317489f303c1059e11843a1b7235177fe2e0c2e6566057b72f7R201

File: miscnn/utils/visualizer.py [hunk 1]
```diff
@@ -267,21 +268,21 @@ def visualize_samples(sample_list, out_dir="vis", mask_seg=False, mask_pred=True
         out_path = os.path.join(out_dir, file_name)
         
         if mask_seg and mask_pred:
-            vol_truth = overlay_segmentation_greyscale(sample.img_data, sample.seg_data)
-            vol_pred = overlay_segmentation_greyscale(sample.img_data, sample.pred_data)
+            vol_truth = overlay_segmentation(sample.img_data, sample.seg_data, sample.classes)
+            vol_pred = overlay_segmentation(sample.img_data, sample.pred_data, sample.classes)
```
Comment (Deathlymad) [inline miscnn/utils/visualizer.py:272]: git shows changes here. this may be due to an odd merge?

**Comment by Deathlymad** [comment]
I just read the github computed diff in the PR tab. But after reading all commits including the merge commit i cannot figure out for the life of me why it says the functions change.

I have not seen a case where the colors act in the way you are describing. Do you haave an exaample case? As for the color spectrum that entirely depends on the colormap that is used. Virdis is the scale commonly used by heatmaps. If you want an ideally distributed scolor spectrum you can just set it to hsv for example.

**Comment by Deathlymad** [comment]
stale, will close for now

**Pull request closed without merging**